regression/abstraction/test2/Readme

```diff
@@ -0,0 +1,4 @@
+Assertions with function "bar" that has hidden index variable.
+We error out currently because functions are not supported inside assertions. 
```

Comment:
If you error out, should this be a `CORE` test? `KNOWNBUG` or `FUTURE` would seem or appropriate unless you are testing the error handling.

Reply:
For the functionality in this PR we don't error out. The readme is referring to the state as it is in our local repo. I will delete the readme files from this PR and introduce them in the next one.

Reply:
Yeah; at the end of each PR things need to be in a consistent state, including documentation and test status.